libdnf/dnf-context.cpp

```diff
@@ -3480,6 +3480,9 @@ dnf_context_module_install(DnfContext * context, const char ** module_specs, GEr
                         std::vector<libdnf::ModuleProfile> profiles;
                         if (resolved_spec.first->getProfile() != "") {
                             profiles = modpkg->getProfiles(resolved_spec.first->getProfile());
+                            if (profiles.empty()) {
+                                throw std::runtime_error("No profile found matching " + resolved_spec.first->getProfile());
```

Comment:
Please use directly gerror.

Reply:
A gerror would mean returning immediately, but I'm trying to match dnf's (and libdnf's other modular functions) behaviour of continuing to process all the inputs before returning errors.

Reply:
I am sorry but there must be some misunderstanding. When you throw, the function stops immediately. But the exception is directly catch by `} CATCH_TO_GERROR(NULL)` and transformed into g_error. Then I suggest to set g_error directly and return.

Reply:
Ahh yes good catch! What I meant was for the exception to be caught by the inner `try`/`catch`, so that it just adds to the `messages` vector, but in copy/pasting I forgot to generalize it to `catch (const std::exception & ex)` instead of just a specific error. I'll change that then. Does that make sense?